### PR TITLE
Adding functionality to confirm before cancelling a review

### LINKED ISSUE #1653: Pull Request summary Cancel button needs confirmation
The Cancel button noted here will remove all of a user's in progress review comments.
The user may not be aware of the effect and may simply be trying to leave the screen because they are not ready to complete their review.

The effect of the cancel button should be clearly explained with a dialog/confirmation before allowing the user to continue.

image

### PR DESCRIPTION
**Pull request by StanleyGoldman**
Fixes #1653

Adding a confirmation to cancel pending reviews as well as a test around it.

### CODE REVIEW

**Comment by StanleyGoldman** [comment]
Shouldn't we stay on the pull request view if the confirmation dialog is cancelled?
I have a branch where I do that, I'm not sure how we feel about the functionality change...

https://github.com/github/VisualStudio/tree/pull-request-dont-close-after-cancel-confirm

@donokuda any feelings?

**Comment by grokys** [comment]
Yeah it does feel to me like nothing should happen if I click "No" in the dialog box, rather than navigating back to the PR.

**Comment by StanleyGoldman** [comment]
I merged that functionality in.

**Comment by meaghanlewis** [comment]
Pressing `Yes` to confirm cancelling a review freezes and restarts Visual Studio for me.

cancel review freezes vs

**Comment by StanleyGoldman** [comment]
Sigh..

File: src/GitHub.InlineReviews/Services/PullRequestSession.cs [hunk 1]
```diff
@@ -223,20 +223,27 @@ public async Task StartReview()
...
-        public async Task CancelReview()
+        public async Task<bool> CancelReview()
         {
             if (!HasPendingReview)
             {
                 throw new InvalidOperationException("There is no pending review to cancel.");
             }
 
-            await service.CancelPendingReview(LocalRepository, PendingReviewId);
+            if (service.ConfirmCancelPendingReview())
```
Comment (grokys) [inline src/GitHub.InlineReviews/Services/PullRequestSession.cs]: Confirming the cancelation should be done in the view model, not the service. The view model should handle user interaction.

File: src/GitHub.App/Services/PullRequestEditorService.cs [hunk 2]
```diff
@@ -288,6 +289,15 @@ public void OpenActiveDocumentInCodeView(IVsTextView sourceView)
             NavigateToEquivalentPosition(sourceView, codeView);
         }
 
+        public bool ConfirmCancelPendingReview()
```
Comment (grokys) [inline src/GitHub.App/Services/PullRequestEditorService.cs]: I'm still not sure this is the right place for this method. The XML doc on the class says:

```
Services for opening views of pull request files in Visual Studio.
```

Which doesn't really fit with this. I'd suggest adding it to `PullRequestSessionService`. I notice you added a `using System.Windows.Forms;` to that file, so I assume you actually started putting it there and then decided not to? Why was that?

Comment (StanleyGoldman) [reply src/GitHub.App/Services/PullRequestEditorService.cs]: - `IPullRequestSessionService` cannot be imported into `PullRequestReviewAuthoringViewModel`
  - `PullRequestReviewAuthoringViewModel` lives in `GitHub.App`
  - `IPullRequestSessionService` lives in `GitHub.InlineReviews`

**Pull request merged**